@here/harp-mapview/lib/geometry/DisplacedMesh.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DisplacementFeature, hasDisplacementFeature } from "@here/harp-materials";
+import { assert } from "@here/harp-utils";
+import * as THREE from "three";
+import { DisplacedBufferAttribute } from "./DisplacedBufferAttribute";
+import { DisplacedBufferGeometry, DisplacementRange } from "./DisplacedBufferGeometry";
+
+function isBufferGeometry(
+    geometry: THREE.Geometry | THREE.BufferGeometry
+): geometry is THREE.BufferGeometry {
+    const isBufferGeom = geometry instanceof THREE.BufferGeometry;
+    assert(isBufferGeom, "Unsupported geometry type.");
+    return isBufferGeom;
+}
+
+function isDisplacementMaterial(material: any): material is DisplacementFeature {
+    const isDisplacementFeature = hasDisplacementFeature(material);
+    assert(isDisplacementFeature, "Material does not support displacement maps.");
+    return isDisplacementFeature;
+}
+
+function isDataTextureMap(map: THREE.Texture | null): map is THREE.DataTexture {
+    const isDataTexture = map !== null && map instanceof THREE.DataTexture;
+    assert(isDataTexture, "Material does not support displacement maps.");
```

Comment:
I guess this one may be checked in runtime (raycast), cause texture may change... Mixins (above) rather not.

Reply:
done.